dl.js

```diff
@@ -1,29 +1,57 @@
 'use strict';
 
 const execSync = require('child_process').execSync;
+module.exports = function dl(version = '3.6.6') {
+  const versionMatch = version.match(/^(\d)\.(\d)\.(\d+)$/);
+  if (!versionMatch) throw new Error('Version must be in x.x.x format');
+  const major = parseInt(versionMatch[1]);
+  const minor = parseInt(versionMatch[2]);
+  const patch = parseInt(versionMatch[3]);
 
-module.exports = function dl(version = '3.6.5') {
-  const os = getOS();
-  const filename = os === 'osx' ?
-    `mongodb-${getOS()}-ssl-x86_64-${version}.tgz` :
-    `mongodb-${getOS()}-x86_64-${version}.tgz`;
-  const dirname = `mongodb-${getOS()}-x86_64-${version}`;
+  let os = process.platform;
+  let dirname;
+  let filename;
 
-  console.log(`Downloading MongoDB ${version}`);
-  execSync(`curl -OL http://downloads.mongodb.org/${getOS()}/${filename}`);
-  execSync(`tar -zxvf ${filename}`);
-  execSync(`mv ./${dirname}/bin ${__dirname}/${version}`);
-  execSync(`rm -rf ./${dirname}`);
-  execSync(`rm ./${filename}`);
-  console.log(`Copied MongoDB ${version} to '${__dirname}/${version}'`);
-};
-
-function getOS() {
-  const os = require('os');
+  switch(os) {
+    case 'linux':
+      filename = `mongodb-linux-x86_64-${version}.tgz`
+      dirname = `mongodb-linux-x86_64-${version}`;
+      break;
+    case 'darwin':
+      os = 'osx'
+      filename = `mongodb-osx-ssl-x86_64-${version}.tgz`;
+      dirname = `mongodb-osx-x86_64-${version}`;
+      break;
+    case 'win32':
+      if (major < 3) {
+        filename = `mongodb-win32-x86_64-2008plus-${version}.zip`;
+        dirname = `mongodb-win32-x86_64-2008plus-${version}`;
+      } else {
+        filename = `mongodb-win32-x86_64-2008plus-ssl-${version}.zip`;
+        dirname = `mongodb-win32-x86_64-2008plus-ssl-${version}`;
+      }
+      break;
+    default:
+      throw new Error(`Unrecognized os ${os}`);
```

Comment:
Just a note for me to fix this in master, this will print out "Unrecognized os [Object object]"

Reply:
@vkarpov15 The output of `process.platform` is string. Is it possible that it returns an object in some cases?

```sh
> typeof process.platform
'string'
> `${process.platform}`
'linux'
```

Reply:
`process.platform` is always a string, but L35 above doesn't use `process.platform`, it uses `os`.

Reply:
Actually, never mind the above comment, you're right, I misread.